### PR TITLE
Guide for AWS Fargate with Vapor and MongoDB Atlas using AWS Copilot

### PR DESCRIPTION
**Pull request by dave-moser**
New guide for deploying Server Side Swift on AWS using AWS Copilot. The guide deploys a RESTful api using Vapor and interacts with MongoDB Atlas to store data.

### CODE REVIEW

**Review by 0xTim — APPROVED** [review]
Couple of minor nits but LGTM overall!

File: docs/aws-copilot-fargate-vapor-mongo.md [hunk 1]
```diff
@@ -0,0 +1,573 @@
...
+import PackageDescription
+
+let package = Package(
+    name: "api",
+    platforms: [
+       .macOS(.v12)
+    ],
+    dependencies: [
+        .package(url: "https://github.com/vapor/vapor", .upToNextMajor(from: "4.7.0")),
+        .package(url: "https://github.com/mongodb/mongodb-vapor", .upToNextMajor(from: "1.1.0-beta.1"))
```
Comment (kmahar) [inline docs/aws-copilot-fargate-vapor-mongo.md:65]: I just noticed we never tagged the GA for 1.1 🤦‍♀️ I'm going to do that this week and then we can update this to not use the beta.

File: docs/aws-copilot-fargate-vapor-mongo.md [hunk 2]
```diff
@@ -0,0 +1,573 @@
...
+```
+
+Initialize a Vapor project named *api*.
+
+```
+vapor new api -n
+```
+
+## Step 3: Add Project Dependencies
+Vapor initializes a *Package.swift* file for the project dependencies. Your project requires an additional library, MongoDBVapor. Add the MongoDBVapor library to the project and target dependencies of your *Package.swift* file.
```
Comment (kmahar) [inline docs/aws-copilot-fargate-vapor-mongo.md:49]:
```suggestion
Vapor initializes a *Package.swift* file for the project dependencies. Your project requires an additional library, [MongoDBVapor](https://github.com/mongodb/mongodb-vapor). Add the MongoDBVapor library to the project and target dependencies of your *Package.swift* file.
```

File: docs/aws-copilot-fargate-vapor-mongo.md [hunk 3]
```diff
@@ -0,0 +1,573 @@
...
+EXPOSE 8080
+
+# Start the Vapor service when the image is run, default to listening on 8080 in production environment
+ENTRYPOINT ["./Run"]
+CMD ["serve", "--env", "production", "--hostname", "0.0.0.0", "--port", "8080"]
+```
+## Step 5: Update the Vapor Source Code
+Vapor also generates the sample files needed to code an API. You must customize these files with code that exposes your To Do List API methods and interacts with your MongoDB database.
+
+The *configure.swift* file initializes an application wide connection to your MongoDB database. It retrieves the connection string to your MongoDB database from an environment variable at runtime.
```
Comment (kmahar) [inline docs/aws-copilot-fargate-vapor-mongo.md:178]:
```suggestion
The *configure.swift* file initializes an application-wide pool of connections to your MongoDB database. It retrieves the connection string to your MongoDB database from an environment variable at runtime.
```
we've seen some confusion in the past from users around whether they need to pool MongoDB clients (they don't - the client manages pooling itself under the hood) so I think it's good to clarify this isn't a single connection

**Review by kmahar — COMMENTED** [review]
this is great!!! few minor comments

**Review by dave-moser — COMMENTED** [review]
Removed the beta tag from the mongodb-vapor dependency import

File: docs/aws-copilot-fargate-vapor-mongo.md (hunk 1 above)
Comment (dave-moser) [reply docs/aws-copilot-fargate-vapor-mongo.md]: Thanks! I have submitted a change to remove the beta tag in the dependency.

**Comment by 0xTim** [comment]
@kmahar as soon as you're happy I'll merge

**Review by kmahar — APPROVED** [review]
LGTM!

**Comment by dave-moser** [comment]
Someone might need to approve my last change/commit - which was fixing the version of the mongo-vapor dependency.

**Comment by 0xTim** [comment]
Think we're good - I'll merge!

**Pull request merged**